nx_cugraph/convert_matrix.py

```diff
@@ -276,3 +284,95 @@ def from_scipy_sparse_array(
     if inplace:
         return create_using._become(G)
     return G
+
+
+@networkx_algorithm(version_added="25.06", fallback=True)
+def to_numpy_array(
+    G,
+    nodelist=None,
+    dtype=None,
+    order=None,
+    multigraph_weight=sum,
+    weight="weight",
+    nonedge=0.0,
+):
+    """MultiGraphs are not yet supported"""
+    # print(" ==> hello!! dispatched to nxcg!")
+    G = _to_graph(G, weight, 1, _get_float_dtype(dtype))
+
+    if nodelist is None:
+        nodelist = list(G)
```

Comment:
Eventually I think we'll want to have `nodelist is None` as a special case that we handle more optimally, since we can use the default ordering of nodes given by `G.src_indices` etc. If `nodelist` is given, then we need to reorder things such that the rows and columns of `A` correspond to the nodes in `nodelist` (so we'll use slightly different `src_indices`).

`test_google_matrix` is a good test to exercise this.